src/integrations/admin/indexing-notification-integration.php

```diff
@@ -147,7 +147,7 @@ public function register_hooks() {
 			\add_action( 'admin_init', [ $this, 'maybe_cleanup_notification' ] );
 		}
 
-		if ( $this->options_helper->get( 'indexing_reason' ) ) {
+		if ( $this->indexing_helper->get_reason() ) {
```

Comment:
A `has_reason` would be more logic at this point.

Reply:
Done, I changed the body of the function to:
`return ( $this->options_helper->get( 'indexing_reason', '' ) !== false )`